File: DTC.Api/Dtos/Board/FinishBoardMatchDto.cs
using System.ComponentModel.DataAnnotations;

namespace DTC.Api.Dtos.Board
{
    public class FinishBoardMatchDto
    {
        [Required]
        public List<BoardParticipantScoreDto> Participants { get; set; } = new();
    }

    public class BoardParticipantScoreDto
    {
        public int ParticipantId { get; set; }

        [Range(0, int.MaxValue)]
        public int Score { get; set; }
    }
}

File: dtc.frontend/src/dtos/board/FinishBoardMatchDto.ts
import z from "zod";

export const boardParticipantScoreDtoSchema = z.object({
  participantId: z.number(),
  score: z.number().int().min(0),
});

export type BoardParticipantScoreDto = z.infer<
  typeof boardParticipantScoreDtoSchema
>;

export const finishBoardMatchDtoSchema = z.object({
  participants: z.array(boardParticipantScoreDtoSchema),
});

export type FinishBoardMatchDto = z.infer<typeof finishBoardMatchDtoSchema>;
